lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyDocumentLinkParticipant.java

```diff
@@ -0,0 +1,63 @@
+package io.openliberty.tools.langserver.lemminx;
+
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import org.apache.xerces.impl.XMLEntityManager;
+import org.apache.xerces.util.URI.MalformedURIException;
+import org.eclipse.lemminx.dom.DOMAttr;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMNode;
+import org.eclipse.lemminx.services.extensions.IDocumentLinkParticipant;
+import org.eclipse.lemminx.utils.XMLPositionUtility;
+import org.eclipse.lsp4j.DocumentLink;
+import org.eclipse.lsp4j.Range;
+
+import io.openliberty.tools.langserver.lemminx.util.LibertyUtils;
+
+
+public class LibertyDocumentLinkParticipant implements IDocumentLinkParticipant {
+
+    private static final Logger LOGGER = Logger.getLogger(LibertyDocumentLinkParticipant.class.getName());
+
+    @Override
+    public void findDocumentLinks(DOMDocument document, List<DocumentLink> links) {
+
+        if (!LibertyUtils.isConfigXMLFile(document))
+            return;
+
+        List<DOMNode> childNodes = document.getDocumentElement().getChildren();
+
+        // collect all <include> nodes that are children of the document element
+        List<DOMNode> includeDomNodes = childNodes.stream().filter(n -> n.getNodeName().equals("include"))
```

Comment:
I feel like there is a lot of similar code here that is also in https://github.com/OpenLiberty/liberty-language-server/blob/main/lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyDiagnosticParticipant.java. Can these two classes share some logic? Maybe put it in LibertyUtils? Or make a new util these two classes utilize?

Reply:
After looking at it again, there are some commonalities, but not enough to make refactoring worthwhile.